pkgs/applications/version-management/git-and-tools/radicle-upstream/default.nix

```diff
@@ -0,0 +1,72 @@
+{ stdenv, appimageTools, gsettings-desktop-schemas, gtk3, patchelf, zlib, glibc, fetchurl }:
+
+let
+  pname = "radicle-upstream";
+  version = "0.1.5";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://releases.radicle.xyz/radicle-upstream-${version}.AppImage";
+    sha256 =  "1q5p6bvzi5awxd9a3xvvdhy26bz0dx8drb1z0zzqdvqqcxxyydq7";
+  };
+
+  contents = appimageTools.extractType2 { inherit name src; };
+
+  git-remote-rad = stdenv.mkDerivation rec {
+    pname = "git-remote-rad";
+    inherit version;
+    src = contents;
+
+    nativeBuildInputs = [ patchelf ];
+    buildInputs = [ zlib glibc ];
+
+    installPhase = ''
+      mkdir -p $out/bin/
+      cp ${contents}/resources/git-remote-rad $out/bin/git-remote-rad
+    '';
+
+    postFixup = ''
+      grr=$out/bin/git-remote-rad
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $grr
+      ORIGIN=$(patchelf --print-rpath $grr)
+      patchelf --set-rpath "${stdenv.lib.makeLibraryPath buildInputs}:$ORIGIN" $grr
+    '';
```

Comment:
You might want to use `autoPatchelfHook` for this. All you need to do is add it to `nativeBuildInputs`, and it would figure out everything for you. It also has the added benefit of erroring out when it finds a missing library dependency.